File: lib/models/user_model.dart
class UserModel {
  String? uid;
  final String email;
  String name;
  final String type;
  String image;
  String image_url;

  UserModel({
    this.uid,
    required this.email,
    required this.name,
    required this.type,
    required this.image,
    required this.image_url,
  });

  Map<String, dynamic> toJson() => {
        'uid': uid,
        'email': email,
        'name': name,
        'type': type,
        'image': image,
        'image_url': image_url,
      };

  Map<String, dynamic> toMap() => {
        'uid': uid,
        'email': email,
        'name': name,
        'type': type,
        'image': image,
        'image_url': image_url,
      };

  factory UserModel.fromMap(Map<String, dynamic> map) => UserModel(
        uid: map['uid'] as String,
        email: map['email'] as String,
        name: map['name'] as String,
        type: map['type'] as String,
        image: map['image'] as String,
        image_url: map['image_url'] as String,
      );

  factory UserModel.fromJson(Map<String, dynamic> json) => UserModel(
        uid: json['_id'] as String?,
        email: json['email'] as String,
        name: json['name'] as String,
        type: json['type'] as String,
        image: json['image'] as String,
        image_url: json['image_url'] as String,
      );
}

File: lib/nodejs/models/user_model.js
const mongoose = require('mongoose');
const bcrypt = require('bcryptjs');

const UserSchema = new mongoose.Schema({
  email: {
    type: String,
    required: true,
    unique: true,
    lowercase: true
  },
  password: {
    type: String,
    required: true
  },
  name: {
    type: String,
    required: true
  },
  type: {
    type: String,
    required: true
  },
  image:{
    type: String,
    required: true
  },
  image_url:{
    type: String,
    required: true
  }
}, {collection: 'users'});

UserSchema.pre('save', async function (next) {
  try {
    const salt = await bcrypt.genSalt(10);
    const hash = await bcrypt.hash(this.password, salt);
    this.password = hash;
    next();
  } catch (e) {
    next(e);
  }
});

UserSchema.methods.isValidPassword = async function (password) {
  try {
    return await bcrypt.compare(password, this.password);
  } catch (e) {
    throw e;
  }
};

module.exports = mongoose.model('User', UserSchema);
